### PR TITLE
[EMCAL-1119] added 6 checkers for the EMC Calibration runs

### PR DESCRIPTION
**Pull request by sweyh99**
6 checkers
Number of Good/Dead/Bad Channels in EMCAL + DCAL
Time Calib Coeff
Fraction of good cells / event
Fraction of good cells / supermodule
Cell amplitude (Calib) vs. supermodule ID PHYS
Cell Time (Calib) vs. supermodule ID (High gain) PHYS

### CODE REVIEW

File: Modules/EMCAL/src/CalibCheck.cxx [hunk 1]
```diff
@@ -0,0 +1,348 @@
...
+      result = Quality::Medium;
+    }
+    else {
+      auto* h_allsupermod_proj = h->ProjectionX("h_supermod_proj");
+      if (h_allsupermod_proj->GetEntries() == 0) {
+        result = Quality::Medium;
+      }
+      else {
+        Double_t mean_allsupermod = h_allsupermod_proj->GetMean();
+        if (mean_allsupermod > mBadThresholdCellTimeSupermoduleCalibPHYS) {
```
Comment (jokonig) [inline Modules/EMCAL/src/CalibCheck.cxx:313]: Shouldn't we also have a lower bound here? The mean can also be shifted to lower times compared to  what we expect

File: Modules/EMCAL/src/CalibCheck.cxx [hunk 2]
```diff
@@ -0,0 +1,348 @@
...
+          result = Quality::Bad;
+        }
+        else if (mean_allsupermod > mMedThresholdCellTimeSupermoduleCalibPHYS) {
```
Comment (jokonig) [inline Modules/EMCAL/src/CalibCheck.cxx:316]: Same as above

File: Modules/EMCAL/src/CalibCheck.cxx [hunk 3]
```diff
@@ -0,0 +1,348 @@
...
+        }
+
+        for (int j=1; j<=h->GetNbinsY(); j++){
+          auto* h_supermod = h->ProjectionY(Form("h_supermod_%d",j),j,j);
+          if (h_supermod->GetEntries() == 0) {
+            result = Quality::Medium;
+          }
+          else {
+            Double_t mean_isupermod = h_supermod->GetMean();
+            if (mean_isupermod > mBadThresholdCellTimeSupermoduleCalibPHYS) {
```
Comment (jokonig) [inline Modules/EMCAL/src/CalibCheck.cxx:327]: same as above

File: Modules/EMCAL/src/CalibCheck.cxx [hunk 4]
```diff
@@ -0,0 +1,348 @@
...
+              result = Quality::Bad;
+              break;
+            }
+            else if (mean_isupermod > mMedThresholdCellTimeSupermoduleCalibPHYS) {
```
Comment (jokonig) [inline Modules/EMCAL/src/CalibCheck.cxx:331]: same as above

File: Modules/EMCAL/src/CalibCheck.cxx [hunk 5]
```diff
@@ -0,0 +1,348 @@
...
+    }
+    else {
+      for (int j=1; j<=h->GetNbinsX(); j++){
+        auto* h_det = h->ProjectionY(Form("h_det_%d",j),j,j);
+        if (h_det->GetEntries() == 0)
+          continue;
+        
+        Float_t avg_frac = 0.0;
+        for (int i=1; i<=h_det->GetNbinsX(); i++){
+          Float_t weight = h_det->GetBinContent(i)*h_det->GetBinCenter(i);
```
[the remark this answers is not in the document]
Comment (sweyh99) [reply Modules/EMCAL/src/CalibCheck.cxx]: Done

File: Modules/EMCAL/src/CalibCheck.cxx [hunk 6]
```diff
@@ -0,0 +1,348 @@
...
+    }
+    else {
+      for (int j=1; j<=h->GetNbinsX(); j++){
+        auto* h_supermod = h->ProjectionY(Form("h_supermod_%d",j),j,j);
+        if (h_supermod->GetEntries() == 0)
+          continue;
+        
+        Float_t avg_frac = 0.0;
+        for (int i=1; i<=h_supermod->GetNbinsX(); i++){
+          Float_t weight = h_supermod->GetBinContent(i)*h_supermod->GetBinCenter(i);
```
[the remark this answers is not in the document]
Comment (sweyh99) [reply Modules/EMCAL/src/CalibCheck.cxx]: Done

File: Modules/EMCAL/src/CalibCheck.cxx [hunk 7]
```diff
@@ -0,0 +1,348 @@
...
+      h_allsupermod_proj->Scale(1.0/h_allsupermod_proj->Integral());
+      for (int i=1; i<=h->GetNbinsY(); i++){
+        auto* h_supermod = h->ProjectionX(Form("h_supermod_%d",i),i,i);
+        if (h_supermod->GetEntries() == 0) {
+          result = Quality::Medium;
+        }
+        else {
+          h_supermod->Scale(1.0/h_supermod->Integral());
+
+          Double_t chi2_SM = -1000;
```
[the remark this answers is not in the document]
Comment (sweyh99) [reply Modules/EMCAL/src/CalibCheck.cxx]: Done

File: Modules/EMCAL/src/CalibCheck.cxx (hunk 2 above)
Comment (sweyh99) [reply Modules/EMCAL/src/CalibCheck.cxx]: Done

File: Modules/EMCAL/src/CalibCheck.cxx (hunk 3 above)
Comment (sweyh99) [reply Modules/EMCAL/src/CalibCheck.cxx]: Done

File: Modules/EMCAL/src/CalibCheck.cxx (hunk 4 above)
Comment (sweyh99) [reply Modules/EMCAL/src/CalibCheck.cxx]: Done

**Review by jokonig — APPROVED** [review]

**Pull request merged**